packages/sdk/android/src/main/java/com/primerioreactnative/components/manager/redirect/PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager.kt

```diff
@@ -0,0 +1,250 @@
+        }
+        promise.resolve(null)
+      }
+
+    }
+
+  }
+
+  @ReactMethod
+  fun onBankSelected(bankId: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.BankId(bankId))
+    promise.resolve(null)
+
+  }
+
+  @ReactMethod
+  fun onBankFilterChange(filter: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.Filter(filter))
+    promise.resolve(null)
+  }
+
+  private suspend fun configureErrorListener() {
+    banksComponent?.componentError?.collectLatest { error ->
+
+      sendEvent(
+        PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_ERROR.eventName,
+        JSONObject().apply {
+          put(
+            "errors", JSONArray(
+              Json.encodeToString(error)
+            )
+          )
+        }
+      )
+    }
+  }
+
+  private suspend fun configureBanksListener() {
+    banksComponent?.componentStep?.collectLatest { banksStep ->
+      when (banksStep) {
+        is BanksStep.Loading -> {
+          sendEvent(
+            PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_RETRIEVING.eventName,
+            JSONObject().apply {
+              put(
+                "retrieving", "true"
+              )
+            }
+          )
+        }
+
+        is BanksStep.BanksRetrieved -> {
+          sendEvent(
+            PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_RETRIEVED.eventName,
+            JSONObject().apply {
+              put(
+                "banks", JSONArray(
+                  Json.encodeToString(banksStep.banks.map { it.toPrimerIssuingBankRN() })
+                )
+              )
+            }
+          )
+        }
+      }
+    }
+  }
+
+  private suspend fun configureValidationListener() {
+    banksComponent?.componentValidationStatus?.collectLatest { validationStatus ->
+      when (validationStatus) {
+        is PrimerValidationStatus.Validating -> {
+          sendEvent(PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_VALIDATING.eventName,
+            JSONObject().apply {
+              put(
+                "validating", "true"
+              )
+            }
+          )
+        }
+
+        is PrimerValidationStatus.Invalid -> {
+          sendEvent(PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_IN_VALID.eventName,
+            JSONObject().apply {
+              put(
+                "errors",
+                JSONArray(
+                  validationStatus.validationErrors.map {
+                    JSONObject().apply {
+                      put(
+                        "errorId", it.errorId
+                      )
+                      put(
+                        "description", it.description
+                      )
+                      put(
+                        "diagnosticsId", it.diagnosticsId
+                      )
+
+                    }
+                  }
+                )
+              )
+            }
+          )
```

Comment:
[nit]: let's create `PrimerValidationErrorRN` as @Serializable and send use it here.